File: src/test/java/com/kiseru/asteroids/server/command/factory/CommandHandlerFactoryTest.java
package com.kiseru.asteroids.server.command.factory;

import com.kiseru.asteroids.server.command.CommandHandler;
import com.kiseru.asteroids.server.command.direction.impl.DownCommandHandler;
import com.kiseru.asteroids.server.command.factory.impl.CommandHandlerFactoryImpl;
import com.kiseru.asteroids.server.command.impl.ExitCommandHandler;
import com.kiseru.asteroids.server.command.impl.GoCommandHandler;
import com.kiseru.asteroids.server.command.impl.IsAsteroidCommandHandler;
import com.kiseru.asteroids.server.command.impl.IsGarbageCommandHandler;
import com.kiseru.asteroids.server.command.impl.IsWallCommandHandler;
import com.kiseru.asteroids.server.command.direction.impl.LeftCommandHandler;
import com.kiseru.asteroids.server.command.direction.impl.RightCommandHandler;
import com.kiseru.asteroids.server.command.impl.UnknownCommandHandler;
import com.kiseru.asteroids.server.command.direction.impl.UpCommandHandler;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import static org.assertj.core.api.Assertions.assertThat;

class CommandHandlerFactoryTest {

    private CommandHandlerFactory underTest;

    @BeforeEach
    void setUp() {
        underTest = new CommandHandlerFactoryImpl();
    }

    @Test
    void testCreatingGoCommandHandler() {
        testCreatingHandler("go", GoCommandHandler.class);
    }

    @Test
    void testCreatingLeftCommandHandler() {
        testCreatingHandler("left", LeftCommandHandler.class);
    }

    @Test
    void testCreatingRightCommandHandler() {
        testCreatingHandler("right", RightCommandHandler.class);
    }

    @Test
    void testCreatingUpCommandHandler() {
        testCreatingHandler("up", UpCommandHandler.class);
    }

    @Test
    void testCreatingDownCommandHandler() {
        testCreatingHandler("down", DownCommandHandler.class);
    }

    @Test
    void testCreatingIsAsteroidCommandHandler() {
        testCreatingHandler("isAsteroid", IsAsteroidCommandHandler.class);
    }

    @Test
    void testCreatingIsGarbageCommandHandler() {
        testCreatingHandler("isGarbage", IsGarbageCommandHandler.class);
    }

    @Test
    void testCreatingIsWallCommandHandler() {
        testCreatingHandler("isWall", IsWallCommandHandler.class);
    }

    @Test
    void testCreatingExitCommandHandler() {
        testCreatingHandler("exit", ExitCommandHandler.class);
    }

    @Test
    void testCreatingUnknownCommandHandler() {
        testCreatingHandler("unknown", UnknownCommandHandler.class);
    }

    private void testCreatingHandler(String command, Class<? extends CommandHandler> expected) {
        var actual = underTest.create(command);

        assertThat(actual).isInstanceOf(expected);
    }
}

File: src/test/kotlin/com/kiseru/asteroids/server/factory/CommandHandlerFactoryTest.kt
package com.kiseru.asteroids.server.factory

import com.kiseru.asteroids.server.command.CommandHandler
import com.kiseru.asteroids.server.command.direction.impl.DownCommandHandler
import com.kiseru.asteroids.server.command.direction.impl.LeftCommandHandler
import com.kiseru.asteroids.server.command.direction.impl.RightCommandHandler
import com.kiseru.asteroids.server.command.direction.impl.UpCommandHandler
import com.kiseru.asteroids.server.command.factory.CommandHandlerFactory
import com.kiseru.asteroids.server.command.factory.impl.CommandHandlerFactoryImpl
import com.kiseru.asteroids.server.command.impl.*
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

internal class CommandHandlerFactoryTest {

    private lateinit var underTest: CommandHandlerFactory

    @BeforeEach
    fun setUp() {
        underTest = CommandHandlerFactoryImpl()
    }

    @Test
    fun testCreatingGoCommandHandler() {
        testCreatingHandler("go", GoCommandHandler::class.java)
    }

    @Test
    fun testCreatingLeftCommandHandler() {
        testCreatingHandler("left", LeftCommandHandler::class.java)
    }

    @Test
    fun testCreatingRightCommandHandler() {
        testCreatingHandler("right", RightCommandHandler::class.java)
    }

    @Test
    fun testCreatingUpCommandHandler() {
        testCreatingHandler("up", UpCommandHandler::class.java)
    }

    @Test
    fun testCreatingDownCommandHandler() {
        testCreatingHandler("down", DownCommandHandler::class.java)
    }

    @Test
    fun testCreatingIsAsteroidCommandHandler() {
        testCreatingHandler("isAsteroid", IsAsteroidCommandHandler::class.java)
    }

    @Test
    fun testCreatingIsGarbageCommandHandler() {
        testCreatingHandler("isGarbage", IsGarbageCommandHandler::class.java)
    }

    @Test
    fun testCreatingIsWallCommandHandler() {
        testCreatingHandler("isWall", IsWallCommandHandler::class.java)
    }

    @Test
    fun testCreatingExitCommandHandler() {
        testCreatingHandler("exit", ExitCommandHandler::class.java)
    }

    @Test
    fun testCreatingUnknownCommandHandler() {
        testCreatingHandler("unknown", UnknownCommandHandler::class.java)
    }

    private fun testCreatingHandler(command: String, expected: Class<out CommandHandler>) {
        val actual = underTest.create(command)
        Assertions.assertThat(actual).isInstanceOf(expected)
    }
}
